source/structured-streaming.txt

```diff
@@ -0,0 +1,327 @@
+
+         For a complete list of methods, see the 
+         `pyspark Structured Streaming reference <https://spark.apache.org/docs/latest/api/python/reference/pyspark.ss.html>`__.
+
+     - id: scala
+       content: |
+
+.. _read-structured-stream:
+.. _continuous-processing:
+
+Configuring a Read Stream from MongoDB
+--------------------------------------
+
+Reading a stream from a MongoDB database requires 
+*continuous processing*, 
+an experimental feature introduced in Spark version 2.3. To learn 
+more about continuous processing, see the `Spark documentation <https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html#continuous-processing>`__.
+
+To use continuous processing with the MongoDB Spark Connector, add the 
+``.trigger()`` method to the ``.writeStream`` of the streaming 
+Dataset or DataFrame that you create from your MongoDB ``.readStream``. 
+In your ``.trigger()``, specify the ``continuous`` parameter.
+
+.. note:: 
+
+   The connector populates its read stream from your MongoDB 
+   deployment's change stream. To populate your change stream, perform 
+   update operations on your database.
+
+   To learn more about change streams, see 
+   :manual:`Change Streams </changeStreams>` in the MongoDB manual.
+
+Specify read stream configuration settings on your local 
+SparkSession ``.readStream``. Lastly, add ``.load()`` to create a read 
+stream with your configuration.
+
+.. note:: 
+   
+   Spark does not begin streaming until you call the ``.start()`` 
+   method on a streaming query.
+
+You must specify the following configuration settings to read from 
+MongoDB:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :widths: 10 40
+
+   * - Setting
+     - Description
+
+   * - ``.readStream.format()``
+     - The format to use for read stream data. Use "mongodb".
+
+   * - ``.writeStream.trigger()``
+     - Enables continuous processing for your ``.readStream``. Use 
+       the ``continuous`` parameter.
+
+.. tabs-drivers::
+
+   tabs:
+     - id: java-sync
+       content: |
+
+     - id: python
+       content: |
+
+         .. code-block:: python
+            :copyable: true
+            :emphasize-lines: 3, 9
+         
+            streamingDataFrame = (<local SparkSession>
+              .readStream
+              .format("mongodb")
+              .load()
+            )
+         
+            query = (streamingDataFrame
+              .writeStream
+              .trigger(continuous="1 second")
+              .format("memory")
+              .outputMode("append")
+            )
+
+            query.start()
+
+         For a complete list of methods, see the 
+         `pyspark Structured Streaming reference <https://spark.apache.org/docs/latest/api/python/reference/pyspark.ss.html>`__.
+
+     - id: scala
+       content: |
+
+Examples
+--------
+
+The following examples show Spark Structured Streaming configurations 
+for streaming between MongoDB and a CSV file.
```

Comment:
Issue: I think csv is formatted like this elsewhere on the page. No preference which way this is formatted, only issue for me is consistency.

```suggestion
for streaming between MongoDB and a `.csv` file.
```

Reply:
Good catch